### PR TITLE
Add back inline draft panel

### LINKED ISSUE #2255: Upload screen in skeleton loader UI
context

### PR DESCRIPTION
**Pull request by jaril**
Fix #2255.

This adds back the draft panel UI. 
- New replays are considered non-initialized
- When the replay author opens a non-initialized Replay, the draft panel in place of the Transcript panel
- When a non-author opens a non-initialized Replay, the Replay appears like usual
- The user can set the replay's title, privacy and/or workspace from the draft panel
- The replay settings are saved when the user presses "Continue"
- The replay is soft-deleted when the user presses "Or discard"
- When the draft screen is displayed, the view toggle is hidden and the recording title simply says "New Recording"

Demo: https://share.descript.com/view/DwVfmp6VmJR

image
image

### CODE REVIEW

**Comment by jaril** [comment]
if replay doesn't have an associated user, skip drafts.

**Comment by jaril** [comment]
Demo: https://share.descript.com/view/ee1aWqYsPEv

**Review by hbenl — COMMENTED** [review]
There are a couple of places where you use a recording that you get from `useGetRecording()` without checking whether it is already available. How do you make sure that it will always be available?

File: src/ui/components/DraftScreen.tsx [hunk 1]
```diff
@@ -15,25 +15,18 @@ function WorkspaceDropdownList({
...
-  }, [workspaces]);
+  const displayedWorkspaces = [{ id: "", key: "", name: "---" }, ...otherWorkspaces];
 
   const onChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
     const selectedId = e.target.value;
     setSelectedWorkspaceId(selectedId);
   };
 
-  if (selectedWorkspaceId == null || displayedWorkspaces.length <= 1) {
+  if (displayedWorkspaces.length == 0) {
```
[the remark this answers is not in the document]
Comment (jaril) [reply src/ui/components/DraftScreen.tsx:29]: Great catch — thanks!

File: src/ui/utils/test.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export function isTest() {
```
[the remark this answers is not in the document]
Comment (jaril) [reply src/ui/utils/test.ts:1]: 👍

**Comment by jaril** [comment]
> There are a couple of places where you use a recording that you get from `useGetRecording()` without checking whether it is already available. How do you make sure that it will always be available?

Added a check for loading in all of those places!

**Pull request merged**